### ISSUE TITLE
Make branch optional

### ISSUE THREAD

**Issue by yelizariev** [stefanzweifel/git-auto-commit-action#27]
**Is your feature request related to a problem? Please describe.**

I mainly use the action to compile code (e.g. rst to html). Which happens in PR branches on pushing, not in pull request. Currently, I either need to hardcode branch (e.g. master), or add an extra step like following:
```
    - run: |
        BRANCH="${GITHUB_REF:11}"
        echo "::set-env name=BRANCH::$BRANCH"
    - name: Commit changes
      uses: stefanzweifel/git-auto-commit-action@v2.4.0
      with:
        branch: ${{ env.BRANCH }}
```

**Describe the solution you'd like**

When not set, the branch should be computed. It could be something like this;

```
BRANCH=${INPUT_BRANCH}
if [[ -z "$INPUT_BRANCH ]]
then
    BRANCH=${GITHUB_REF:11}
fi

```

**Comment by stefanzweifel** [comment]
We can surely try to add some code which computes the correct branch name.

We [used to have](https://github.com/stefanzweifel/git-auto-commit-action/blob/v2.0.0/entrypoint.sh#L29) `${GITHUB_REF:11}` in the original scripts. Some changes made by GitHub during the beta broke that behaviour and we added the `branch`-argument instead.

I also remember that I caused infinite loops using `${GITHUB_REF:11}`. Workflows would on the fly create new branches and would start other workflow runs. Only disabling the repository stopped Actions.

But as you're already using similar code in your projects, I think it would make sense to add this behaviour make as a fallback, when no branch name has been given.

**Comment by stefanzweifel** [comment]
I've just released v3.0.0 of the Action. The branch name is now optional (still required if the Workflow is listening to the `pull_request`-event though 😞).

The changes have been tested with Workflows which listen to the `push`-event (like you mentioned in your opening comment). 
There's a usage example in the [README](https://github.com/stefanzweifel/git-auto-commit-action#example-usage).

I'm closing this issue now, as I see this issue as resolved now. Feel free to reopen this issue or create a new issue.
